external/netutils/netlib_netmon.c

```diff
@@ -0,0 +1,170 @@
+ ****************************************************************************/
+
+#if defined(CONFIG_NET_IPv4)
+#define PF_INETX PF_INET
+#elif defined(CONFIG_NET_IPv6)
+#define PF_INETX PF_INET6
+#endif
+#define NUM_SOCKETS CONFIG_NSOCKET_DESCRIPTORS
+
+#ifdef CONFIG_NET_IPv6								
+#define PRINT_IPV(type)						\
+	do {											\
+		if (type & NETCONN_TYPE_IPV6) {				\
+			printf("IPv6\t");						\
+		} else {									\
+			printf("IPv4\t");						\
+		} 											\
+	} while (0)
+#else
+#define PRINT_IPVERSION(type)						\
+	do {											\
+		printf("IPv4\t");							\
+	} while (0)
+#endif
+
+
+/**
+ * Print a externally used socket information.
+ */
+static void print_socket(FAR struct socket *sock)
+{
+	int i;
+	int ret_chk;
+	char get_name[32];
+
+	printf("\nSockfd\tProto\tType\tLocal\t\t\tRemote\t\t\tstate\tPID\n");
+
+	for (i = 0; i < NUM_SOCKETS; ++i) {
+	/* Protect socket array */
+		if (sock[i].conn) {
+			printf("%-8d",i);
+			/* Protocol */
+			if (sock[i].conn->type & NETCONN_TCP) {
+				printf("TCP\t");
+				PRINT_IPV(sock[i].conn->type);
+				printf("%-15s:%-8s", inet_ntoa(sock[i].conn->pcb.ip->local_ip),"*");
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->remote_ip), sock[i].conn->pcb.tcp->remote_port);
+			} else if (sock[i].conn->type & NETCONN_UDP) {
+				printf("UDP\t");
+				PRINT_IPV(sock[i].conn->type);
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->local_ip),sock[i].conn->pcb.udp->local_port);
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->remote_ip), sock[i].conn->pcb.udp->remote_port);
+			} else {
+				printf("RAW\t");
+				PRINT_IPV(sock[i].conn->type);
+				printf("%-15s:%-8s", inet_ntoa(sock[i].conn->pcb.ip->local_ip),"*");
+				printf("%-15s:%-8s", inet_ntoa(sock[i].conn->pcb.ip->remote_ip),"*");
+			}
+
+			switch (sock[i].conn->state) {
+				case NETCONN_WRITE:
+					printf("WRITE\t");
+					break;
+				case NETCONN_LISTEN:
+					printf("LISTEN\t");
+					break;
+				case NETCONN_CONNECT:
+					printf("CONNECT\t");
+					break;
+				case NETCONN_CLOSE:
+					printf("CLOSE\t");
+					break;
+				default:
+					printf("NONE\t");
+					break;
+			}
+			ret_chk = pthread_getname_np(sock[i].conn->pid, get_name);
+			if (ret_chk == OK) {
+				printf("%d:%s\n", sock[i].conn->pid, get_name);
+			} else {
+				printf("%d:NONE\n", sock[i].conn->pid);
+			}
+		} 
+	}
+	return ;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: netlib_netmon
+ *
+ * Description:
+ *   Get the network driver IPv4 address
+ *
+ * Parameters:
+ *   command   Type of information to get
+ *
+ * Return:
+ *   0 on success; -1 on failure
+ *
+ ****************************************************************************/
+
+int netlib_netmon(FAR const char *command)
```

Comment:
I think using command in function is not good. command have to be used application. so please change API.

Reply:
I understand your point. Change will be appiled.